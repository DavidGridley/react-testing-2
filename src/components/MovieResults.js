import React from "react";
import MovieResult from "./MovieResult";

function MovieResults({ movies }) {
  return (
    <div>
      <ul>
        {movies.map(movie => (
          <MovieResult key={movie.imdbID} movie={movie} />
        ))}
      </ul>
    </div>
  );
}

export default MovieResults;
