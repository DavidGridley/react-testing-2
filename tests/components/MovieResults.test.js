import React from "react";
import MovieResults from "../../src/components/MovieResults";
import renderer from "react-test-renderer";

describe("MovieResults", () => {
  test("matches the snapshot", () => {
    const tree = renderer
      .create(
        <MovieResults
          movies={[
            { Title: "pulp fiction", Year: "1995", Poster: "N/A", imdbID: "1" },
            { Title: "another film", Year: "1996", Poster: "N/A", imdbID: "2" },
            { Title: "another film", Year: "1996", Poster: "N/A", imdbID: "3" }
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
