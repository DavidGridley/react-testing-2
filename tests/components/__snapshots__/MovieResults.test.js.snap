// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieResults matches the snapshot 1`] = `
<div>
  <ul>
    <li>
      <h2>
        pulp fiction
         - 
        1995
      </h2>
      <div>
        <span>
          Movie has 
          0
           likes
        </span>
        <button
          onClick={[Function]}
        >
          Like
        </button>
      </div>
      <img
        src="N/A"
      />
    </li>
    <li>
      <h2>
        another film
         - 
        1996
      </h2>
      <div>
        <span>
          Movie has 
          0
           likes
        </span>
        <button
          onClick={[Function]}
        >
          Like
        </button>
      </div>
      <img
        src="N/A"
      />
    </li>
    <li>
      <h2>
        another film
         - 
        1996
      </h2>
      <div>
        <span>
          Movie has 
          0
           likes
        </span>
        <button
          onClick={[Function]}
        >
          Like
        </button>
      </div>
      <img
        src="N/A"
      />
    </li>
  </ul>
</div>
`;
